### ISSUE TITLE
apidoc: Add a -M option to put module documentation before submodule documentation

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#1456]
For some use cases, depth-first display in API documentation makes sense. The code paths are:

```
Module
-> Submodule
-> Submodule
```

So it makes sense to have the generated documentation be

```
Module
-> Submodule
-> Submodule
```

Whereas currently, apidoc generated documentation pages are:

```
-> Submodule
-> Submodule
Module
```

Patch by Luc Saffre originally described here: https://groups.google.com/forum/#!topic/sphinx-users/Bm4w7OQcWYM
- Omits the 'Module contents' heading (which seems implicit)
- Puts 'master_package' documentation in the page header

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/1456
- Originally reported by: Wes Turner
- Originally created at: 2014-04-20T02:06:53.601

**Comment by shimizukawa** [comment]
_From Wes Turner on 2014-04-20 00:10:54+00:00_

Pull request with -M option added here: https://bitbucket.org/birkenfeld/sphinx/pull-request/236/1456-apidoc-add-a-m-option-to-put-module/diff

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2014-05-05 03:23:27+00:00_

fix: For a apidoc -M feature, packages with empty docstring caused a syntax error. Closes #1456.

# see also https://bitbucket.org/birkenfeld/sphinx/pull-request/236

→ <<cset 7a44d4d0ebe99f529216ba57ae26de25687cb135>>

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2014-05-05 03:24:29+00:00_

Thanks!

**Comment by shimizukawa** [comment]
_From Wes Turner on 2014-05-05 11:40:04+00:00_

Thanks!

Wes Turner

**Comment by shimizukawa** [comment]
_From Arve Seljebu on 2014-11-21 15:54:06+00:00_

Is this removed?

```
#!bash
$ sphinx-apidoc --version
Sphinx (sphinx-apidoc) 1.2.3
```

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2014-11-21 23:37:54+00:00_

[Arve Seljebu](https://bitbucket.org/arve0) No. `--version` option for sphinx-apidoc is still exist. something wrong?

**Comment by shimizukawa** [comment]
_From Arve Seljebu on 2014-11-21 23:40:37+00:00_

Sorry, I was not explicit: the `-M` option is not available in apidoc version 1.2.3. Is this right?

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2014-11-21 23:45:20+00:00_

Ah, ok. This feature will be shipped from 1.3 and it is contained in Sphinx-1.3b1.

**Comment by shimizukawa** [comment]
_From Arve Seljebu on 2014-11-21 23:46:38+00:00_

Ok, thanks :-)
